### ISSUE TITLE
Call for help Iranians to access free Internet اعلامیه درخواست کمک از ایرانیان خارج کشور جهت کمک به ارائه اینترنت به مردم

### ISSUE THREAD

**Issue by free-the-internet** [net4people/bbs#202]
From September 2022 until now, the people of Iran are under the heaviest censorship and their access to Free Internet have been extremely limited by the regime. Below is a description of the current situation:
* Almost all free and commercial VPNs are blocked
* SSH and many other tunneling services are blocked or heavily throttled or interrupted after a few seconds
* Many TLS based tunneling services are blocked by means of various methods/rules that some of them are known, and many are unknown for us
* Only Tor Snowflake technology as a public anti censorship tool is working thanks to the volunteers, but just in some regions
* Blocking and limiting is localized and different between regions, operators, and type of the service

These days that the streets are not full of protesters, the censor is taking the next steps to tighten the access of the people. There is multiple news out in the recent days that the regime requested to list the name of the selected organizations or persons who wants to access to Internet, and then probably close the Internet to the public.
Unfortunately, the economy corresponding or relying to international IT services are destroyed in the recent years, and the regime is ready more than before to shutdown the Internet; since the regime does not feel pressures any more!

This pushed many of the Iranian abroad, who have the required knowledge, to maintain their connection with the family and friends by means of the purchasing personal servers and using one of the public tools. While Tor remained the most available and reliable anti-censor tool for the public.

I said all these to call for help from Iranians abroad, specially the very talented and specialized ones to be united against the censor, and create a solid group to provide proxy tools and services to all of Iranians inside the country to help them to keep their connection to outside.
Of course, we also need the international organizations who are active in "defending the human rights" and "providing Internet access". We knew that, there are already many organization for analyzing and monitoring the connectivity, but there is not much effort to provide and build tools fit the specific country situations.

For this to happen, we need volunteers in various field to manage our organization:
* Network engineers / Cloud developers / Security engineers
* Server administrators
* Financial mangers
* Lawyers
* Political activists
* Independent organization to collect the donations

We need "Financial mangers" and "Lawyers" to help us to get and manage the contract and costs for services from big firms like AWS, and many other big and small companies to provide IP diversity to avoid mass blockage of a specific IP range.
We need "Political activists" and "Independent organization" to promote our activity, advertise the activity, and collect the funds. A well known Independent organization is very important for donors to trust and make sure that their donation is helpful.
I believe that the identity of the engineers could remain unknown as the tasks should be done in a git. But the political activists and the independent organization must be publicly known. I also hope we can have access to financial managers and lawyers through the independent organization.

Please let me know what do you think and how we can proceed?
I would be very thankful if you share and send this thread in all social networks to be seen by the volunteers.

از سپتامبر 2022 تاکنون، مردم ایران تحت شدیدترین سانسور هستند و دسترسی آنها به اینترنت رایگان توسط حکومت به شدت محدود شده است. در زیر شرحی از وضعیت فعلی آورده شده است:
* تقریباً تمام وی پی ان های رایگان و تجاری مسدود شده اند
* SSH و بسیاری دیگر از سرویس های تونل زنی مسدود شده یا به شدت تحت محدودیت قرار می گیرند یا پس از چند ثانیه قطع می شوند.
* بسیاری از خدمات تونل زنی مبتنی بر TLS با استفاده از روش ها/قوانین مختلفی مسدود می شوند که برخی از آنها شناخته شده هستند و بسیاری از آنها برای ما ناشناخته هستند.
* تنها فناوری Tor Snowflake به عنوان یک ابزار عمومی ضد سانسور به لطف داوطلبان کار می کند، اما فقط در برخی مناطق
* مسدود کردن و محدود کردن محلی است و بین مناطق، اپراتورها،و نوع خدمات متفاوت است

این روزها که خیابان ها مملو از معترض نیست، سانسور گام های بعدی را برای تشدید محدودیت دسترسی مردم به اینترنت برمی دارد. در روزهای اخیر اخبار متعددی منتشر شده است مبنی بر اینکه حکومت درخواست کرده اسامی سازمان ها یا افرادی که می خواهند به اینترنت دسترسی داشته باشند را فهرست کند و احتمالاً اینترنت را به روی عموم ببندد.
متأسفانه اقتصاد منطبق یا متکی به خدمات بین‌المللی فناوری اطلاعات در سال‌های اخیر در ایران از بین رفته است و حکومت بیش از پیش آماده است تا اینترنت را قطع کند. چون حکومت دیگر فشاری را احساس نمی کند!

این امر باعث شد بسیاری از ایرانیان خارج از کشور که دانش لازم را دارند، با خرید سرورهای شخصی و استفاده از یکی از ابزارهای عمومی ارتباط خود را با خانواده و دوستان حفظ کنند. در حالی که ابزار «تور» در دسترس ترین و قابل اعتمادترین ابزار ضد سانسور برای عموم باقی مانده است.

اما همه اینها را گفتیم تا از ایرانیان خارج از کشور کمک بخواهیم، مخصوصاً افراد بسیار با استعداد و متخصص که علیه سانسور متحد شوند و یک گروه مستحکم ایجاد کنند تا ابزارها و خدمات پروکسی را به همه ایرانیان داخل کشور ارائه دهند و به آنها کمک کنند تا اتصال به اینترنت را حفظ کنند.
البته ما به سازمان های بین المللی نیز نیاز داریم که در «دفاع از حقوق بشر» و «تامین دسترسی به اینترنت» فعال باشند. ما می‌دانیم که سازمان‌های زیادی برای تجزیه و تحلیل و نظارت بر وضعیت اتصال اینترنت وجود دارد، اما تلاش زیادی برای تهیه و ساخت ابزار متناسب با شرایط خاص کشور وجود ندارد.

برای تحقق این امر، ما به داوطلبانی در زمینه های مختلف برای مدیریت سازمان خود نیاز داریم:
* مهندسان شبکه / توسعه دهندگان کلاود / مهندسین امنیت
* مدیران سرور
* مدیران مالی
* وکلا
* فعالان سیاسی
* سازمان مستقل برای جمع آوری کمک ها

ما به "مدیران مالی" و "وکلا" نیاز داریم تا به ما کمک کنند تا قرارداد و هزینه های خدمات را از شرکت های بزرگی مانند آمازون و بسیاری از شرکت های بزرگ و کوچک دیگر دریافت و مدیریت کنیم تا تنوع آی‌پی  را برای جلوگیری از مسدود شدن انبوه یک محدوده آی‌پی خاص ارائه دهیم.
ما به "فعالان سیاسی" و "سازمان مستقل" برای ترویج فعالیت، تبلیغ فعالیت و جمع آوری کمک نیازمندیم. یک سازمان مستقل شناخته شده برای اهداکنندگان بسیار مهم است که به آن اعتماد کنند و مطمئن شوند که کمک مالی آنها مفید است.

ما معتقدیم که هویت مهندسان می تواند ناشناخته بماند زیرا وظایف می تواند به صورت یکپارچه در git انجام شود. اما فعالان سیاسی و تشکل مستقل باید در سطح عمومی شناخته شوند. همچنین امیدوارم بتوانیم از طریق سازمان مستقل به مدیران مالی و وکلا دسترسی داشته باشیم.

لطفاً نظر خود را به ما بگویید و اینکه چگونه می توانیم شروع کنیم؟
اگر این موضوع را در تمام شبکه های اجتماعی به اشتراک بگذارید و ارسال کنید تا توسط داوطلبان دیده شود بسیار سپاسگزار خواهم بود.

**Comment by free-the-internet** [comment]
Please reply to this thread only if:
* you have ideas and planing
* you want to become a volunteer
* you know any organization/well known person that can help

**Comment by markpash** [comment]
I'm happy to help, if my skill-set is required.

**Comment by free-the-internet** [comment]
> I'm happy to help, if my skill-set is required.

Sure. We want people for scripting to make all the tasks automatic. So we need people who have experience with APIs to create and manage instances and scale the resources when needed.
For now, I only worried about the IPs and the fact that they should change frequently; and I don't know if all the providers has the possibility to manage the IP automatically. Specially the small ones.

**Comment by sambali9** [comment]
I am also happy to help for the network and scripting part.

As you mentioned we need a lot of IPs as they get banned rapidly when the traffic goes up. My Idea is to use public VPNs IPs for our servers. It is technically possible with nat traversal (the same thing Tor's snowflake is doing). That way all we need are a few central instances which connect to public VPNs (or proxies) and the clients in Iran can have a p2p connection to the VPNs' IPs. This will need a stun server but this can be solved with domain fronting (just like snowflake). The difference with snowflake is that it won't go through Tor which would massively improve latency and bandwidth.

I have other ideas but this one I think is a good starting point (in theory at least).

**Comment by mehdifirefox** [comment]
تجربه موفقی که اینروزها درحال انجام راه اندازی ربات های تلگرام  با مثلا حجم هفتگی هست که راه افتاده 

این ایده عالیه برای افرادی که میخوان کمکی کنن ولی راهی پیدا نمیکنن چطور به دست کاربرا برسونن

**Comment by free-the-internet** [comment]
I would be thankful if you share this thread in twitter and/or with the persons you guess potentially they can help,
Of course, if you have any contact with the independent organizations, please share.
I see the in recent months, some Iranian independent organizations started to appear in western countries. These organizations have a good list of the well known Iranian politicians, musicians and others who want help their country. If you have any contact with them, please share to speed up building this community.
Without any oversea organization that can get donations and do the management we will never succeed.
Also, you can mention the name of these kind of organizations if you knew any.

سپاسگزار خواهم بود اگر این موضوع را در توییتر و یا با افرادی که حدس می‌زنید می‌توانند کمک کنند به اشتراک بگذارید،
البته اگر با تشکل های مستقل ارتباطی دارید حتما به اشتراک بگذارید که شدیا به کمک آنها نیازمندیم.
من می بینم که در ماه های اخیر برخی از سازمان های مردمی و مستقل ایرانی در کشورهای غربی شروع به ظهور کردند. این سازمان ها فهرست خوبی از سیاستمداران، موسیقیدانان و سایرین سرشناس ایرانی دارند که می خواهند به کشورشان کمک کنند. اگر با آنها تماس دارید، برای تسریع در ساخت این انجمن به اشتراک بگذارید. میتوانید به آنها ایمیل بزنید و یا پیام بفرستید و از یشان دعوت کنید.
بدون هیچ سازمان خارجی (خارج از ایران) مستقل که بتواند کمک های مالی دریافت کند و مدیریت را انجام دهد، ما هرگز موفق نخواهیم شد.
اگر از این دسته از سازمان ها میشناسید، اینجا اعلام کنید.

**Comment by arandomgstring** [comment]
> These days that the streets are not full of protesters, the censor is taking the next steps to tighten the access of the people. There is multiple news out in the recent days that the regime requested to list the name of the selected organizations or persons who wants to access to Internet, and then probably **close the Internet to the public.**

If they do this, nothing can be done, no, really. Because circumvent tools ultimately work on software level. If ISPs decide to whitelist access of users to certain IPs, with say, a few IPtable rules, no software can bypass this. Even If we use IP spoofing to pretend that a proxy is a whitelisted website, still we cannot send any information to proxy server, because we have to use proxy's real IP which is not whitelisted. At that point, only hardware solutions such as satellite internet can be used. Therefore I really don't understand the point of asking help from software/network engineer for this scenario, nor do I think lawyers (be in Iran or outside of it) can do anything about this. 

Therefore, the only feasible idea that I can think of, is to make satellite internet as humanly common as possible in Iran. Donations should be well spent on something like this, I guess. Foreign governments with this technology should be convinced to invest in Iran. But again, because Rial is worthless, and changing Rial to other currencies while fighting against money laundering is challenging to say the least, I don't think that this investment would be fruitful. In the end, It boils down to people's reaction to Internet shutdown; which I think we have already seen it to some extent.

**Comment by free-the-internet** [comment]
> > These days that the streets are not full of protesters, the censor is taking the next steps to tighten the access of the people. There is multiple news out in the recent days that the regime requested to list the name of the selected organizations or persons who wants to access to Internet, and then probably **close the Internet to the public.**
> 
> If they do this, nothing can be done, no, really. Because circumvent tools ultimately work on software level. If ISPs decide to whitelist access of users to certain IPs, with say, a few IPtable rules, no software can bypass this. Even If we use IP spoofing to pretend that a proxy is a whitelisted website, still we cannot send any information to proxy server, because we have to use proxy's real IP which is not whitelisted. At that point, only hardware solutions such as satellite internet can be used. Therefore I really don't understand the point of asking help from software/network engineer for this scenario, nor do I think lawyers (be in Iran or outside of it) can do anything about this.
> 
> Therefore, the only feasible idea that I can think of, is to make satellite internet as humanly common as possible in Iran. Donations should be well spent on something like this, I guess. Foreign governments with this technology should be convinced to invest in Iran. But again, because Rial is worthless, and changing Rial to other currencies while fighting against money laundering is challenging to say the least, I don't think that this investment would be fruitful. In the end, It boils down to people's reaction to Internet shutdown; which I think we have already seen it to some extent.

Thanks for your comment.
Indeed this call is for providing Internet to the people while there is no complete shutdown.
I think after the community have been established, we can decide to whether to follow the path to connect with the people who can lobby for providing satellite Internet to the people of Iran in a large scale or not. We know that they did for Ukraine; but this is another story and topic.

Now, we really focus on providing the circumvent tools in a large scale to the public.
I would amend the first post ASAP a bit to eliminated this misunderstanding.

**Comment by poorp** [comment]
I have an idea that is based on existing software but needs a lot of programming to happen.
Currently there is a specific tunneling method that works and I prefer not to name. Only problem is it needs a domestic "Iran server" which is a really bad idea.
What I propose is to eliminate the need for an Iran server and I know that it can be done by developing a certain app for Android/IOS (the pc app already exists and works but it's CLI and there is no GUI for it so we could perhaps make a GUI for that as well)
This project needs a mobile back-end programmer familiar with networking and internet connections and protocols and a front-end programmer and perhaps maybe a UI designer to make GUI possible for it.
If and only if you can provide said resources, contact me again via replying to this so we can get in touch about it.

**Comment by VZiChoushaDui** [comment]
Hi,

We at [Libertea](https://github.com/VZiChoushaDui/Libertea) are committed to provide easy-to-setup, reliable and scalable tools for censorship circumvention. Our vision is to make it as easy as possible, so many people with minimal technical knowledge can set up their own censorship circumvention solution.

We seem to have quite a few users in Iran as well; we're happy to receive feedback/analysis/discussion about how Libertea is working in Iran in order to improve its reliability there and optimize it for the situation there; and of course we welcome contributions.

**Comment by free-the-internet** [comment]
I see Libertea, and also some other initiatives like getafreenode.com, and many other repositories and others activists working hard in GitHub publicly to make new tools for deploying TLS based VPNs.
While this is admirable, but there is still 3 important problems:
1. Ordinary people still can not afford individual server costs
2. Ordinary people do not have enough information to deploy (even there are many tools to do it, but imagine the people who doesn't heard even about SSH, server, client, cloud, etc)
3. These solutions do not scale.

So we have very big problems that requires contribution of the independent organizations.
Until now, I observe that the only voluntaries are at least Software developers or Network engineers; but we need more people that are politically active and they can raise our voice and lobby for getting services.

من می بینم که Libertea، و همچنین برخی ابتکارات دیگر مانند getafreenode.com، و بسیاری از مخازن دیگر و سایر فعالان در GitHub به طور عمومی سخت کار می کنند تا ابزارهای جدیدی برای استقرار VPN های مبتنی بر TLS ایجاد کنند.
اگرچه این قابل تحسین است، اما هنوز 3 مشکل مهم وجود دارد:
1. مردم عادی هنوز هم نمی توانند هزینه های سرور فردی را بپردازند
2. افراد عادی اطلاعات کافی برای استقرار ندارند (حتی ابزارهای زیادی برای انجام آن وجود دارد، اما افرادی را تصور کنید که حتی در مورد SSH، سرور، کلاینت، ابر و غیره چیزی نشنیده اند)
3. این راه حل ها در مقیاس بزرگ قابل ارائه نیستند.

بنابراین ما مشکلات بسیار بزرگی داریم که نیازمند مشارکت سازمان های مستقل است.
تا به حال، من مشاهده می کنم که تنها داوطلبان، حداقل (از نظر دانش) توسعه دهندگان نرم افزار یا مهندسان شبکه هستند. اما ما به افراد بیشتری نیاز داریم که از نظر سیاسی فعال باشند و بتوانند صدای ما را بلند کرده و برای دریافت خدمات لابی کنند.

**بنابراین لازم است که اگر طرفدار گروه داوطلبی هستید، یا جزو رهبران یا جزو نزدیکان به رهبران گروه های سیاسی مخالف هستید، حتما این جستار را به آنها معرفی کنید که با مشارکت آنها بشود یک تشکلی ایجاد گردد که برای حل مشکلات بالا وارد عمل شود.**

سپاسگذارم.

**Comment by BlizzardMulholland** [comment]
a

**Comment by operationairstrike** [comment]
Hey I'm good at networking and cyber security XD. If I can help give me a contact.

**Comment by operationairstrike** [comment]
> > I'm happy to help, if my skill-set is required.
> 
> Sure. We want people for scripting to make all the tasks automatic. So we need people who have experience with APIs to create and manage instances and scale the resources when needed.
> For now, I only worried about the IPs and the fact that they should change frequently; and I don't know if all the providers has the possibility to manage the IP automatically. Specially the small ones.
> 
> از همه ی اهالی فن که در این مورد اطلاعاتی دارند، خواهش میکنم که در اینجا موضوعات را به صورت خیلی خلاصه بگویند و روشن کنند. مخصوصا جزئیات فنی باید بعدا در جای خود بحث بشود. فعلا در اینجا دنیال داوطلبانی هستیم که بتوانند برای شروع کمک کنند.
> ترجیحا افراد دارای تجربه باشند که خیلی فوری بتوانیم کار را شروع کنیم.

Hey I'm good at node.js apis

**Comment by woodlyer** [comment]
I can give you some technology tips.

**Comment by free-the-internet** [comment]
> سلام، میشه بیشتر توضیح بدید که اینترنت بسته بشه چطور ممکنه بسته بشه، آیا بعد ها با VPN میشه رفع کرد یا خیر، کلا VPN هم نمیتونه کاری بکنه؟

اگر اینرتنت بسته شود، و یک شبکه ی داخلی راه اتدازی شود، تقریبا هیچ وی پی ان ای دیگر کار نخواهد کرد. مگر اینکه یک راه عبور اطلاعات با خارج پیدا شود (مانند خریدن سرورهای کلاود داخلی که ممکن است به اینرتنت جهانی وصل باشند). یا با انواع روشهایی مانند خریدن سیم کارت های مخصوص توریست ها یا استاد دانشگاه یا ... که هیچ سودی برای هیچ کسی به غیر از رژیم جنایت کار ندارد. البته گر نیک بنگرند، برای خودشان هم در دراز مدت هیچ سودی نخواهد داشت!

**Comment by cinascorp** [comment]
سلام ، اخیرا گیت هاب در ایران باز شده ، از ایرانیان مقیم خارج از کشور خواهش میکنیم صدای ما باشید ، اینجا فقط به سایت ها با دامنه 
سایت.ir 
لطفا پورت 53 هاست و دامنه خودتون رو برای اشتراک باز بزارید و به ما اجازه بدید از 
dns
 شما استفاده کنیم ، این کار نه برای شما هزینه اور هست و نه خطرناک
کانفیگ هاتون رو به 
Base64 
تبدیل کنید ، و اون رو با عزیزانتون به اشتراک بگزارید .
شما لایق بهترین ها هستید ❤️
جانم فدای ایران .
مرگ بر رژیم فاسد آخوندی 

*Hello, GitHub has recently been unblocked in Iran. We ask Iranians living abroad to be our voice. Here, only sites with a .ir domain are affected. Please open port 53 on your host and domain for sharing and allow us to use your DNS. This is neither costly nor dangerous for you.
Convert your config to Base64 and share it with your loved ones.
You deserve the best ❤️
My life for Iran.
Death to the corrupt clerical regime.*

**Comment by free-the-internet** [comment]
> سلام ، اخیرا گیت هاب در ایران باز شده ، از ایرانیان مقیم خارج از کشور خواهش میکنیم صدای ما باشید ، اینجا فقط به سایت ها با دامنه سایت.ir لطفا پورت 53 هاست و دامنه خودتون رو برای اشتراک باز بزارید و به ما اجازه بدید از dns شما استفاده کنیم ، این کار نه برای شما هزینه اور هست و نه خطرناک کانفیگ هاتون رو به Base64 تبدیل کنید ، و اون رو با عزیزانتون به اشتراک بگزارید . شما لایق بهترین ها هستید ❤️ جانم فدای ایران . مرگ بر رژیم فاسد آخوندی
> 
> _Hello, GitHub has recently been unblocked in Iran. We ask Iranians living abroad to be our voice. Here, only sites with a .ir domain are affected. Please open port 53 on your host and domain for sharing and allow us to use your DNS. This is neither costly nor dangerous for you. Convert your config to Base64 and share it with your loved ones. You deserve the best ❤️ My life for Iran. Death to the corrupt clerical regime._

درود
لطفا اطلاعات بیشتری در اینجا به اشتراک بگذارید. چه چیزی باید روی پورت 53 اجرا کنند و اطلاعات دیگر.

*Hello
Please share more information here. What should be running on port 53 and other details.*
